### DISCUSSION TITLE
Respect SSH_ASKPASS

### DISCUSSION THREAD

**Discussion by lmtr0** [gitbutlerapp/gitbutler#5373] [Q&A]
Hello there, I hope you are ok!
I'm having a slight annoyance. I have to re-enter my ssh password every time I have to push something to the remote repo. I use the System Keyring to store them when I use git itself.

How can I configure Git Butler to respect SSH_ASKPASS env variable?

**Comment by Byron** [comment]
Thanks for asking!

The problem is usually the difference between terminals and environment variables in login shells, and GUI applications.
You can see if this is indeed the case here by launching GitButler from a terminal window using it's whole path, like `/Applications/GitButler.app/Contents/MacOS/GitButler` on MacOS for example.

It's possible that this doesn't work though as GitButler also overrides this value with its own:

```
gitbutler ( master) [$] via 
❯ rg SSH_ASKPASS
crates/gitbutler-git/src/repository.rs
141:    envs.insert("SSH_ASKPASS".into(), askpath_path);
143:    // DISPLAY is required by SSH to check SSH_ASKPASS.
```

In case it doesn't work, we should probably turn this question into an issue and see if it's possible to get this working anyway.

**Reply by lmtr0** [reply]
Didn't work. I guess we have to add a option for it in the settings and allow Git Butler to respect the system settings. I'll work on this later this afternoon, meanwhile I'm opening a issue
